File: services/core/src/main/kotlin/com/innorder/occ/system/SystemStatus.kt
package com.innorder.occ.system

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonFormat
import java.time.Instant

enum class ServiceState {
    READY,
    DEGRADED,
    UNREACHABLE,
    CHECKING,
}

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ComponentStatus(
    val id: String,
    val label: String,
    val state: ServiceState,
    val detail: String? = null,
    @field:JsonFormat(shape = JsonFormat.Shape.STRING)
    val checkedAt: Instant,
)

data class SystemStatus(
    val service: String,
    val version: String,
    val state: ServiceState,
    @field:JsonFormat(shape = JsonFormat.Shape.STRING)
    val checkedAt: Instant,
    val components: List<ComponentStatus>,
)

File: packages/contracts/src/system-status.ts
import { z } from "zod";

export const ServiceStateSchema = z.enum([
  "READY",
  "DEGRADED",
  "UNREACHABLE",
  "CHECKING",
]);

export type ServiceState = z.infer<typeof ServiceStateSchema>;

export const ComponentStatusSchema = z
  .object({
    id: z.string(),
    label: z.string(),
    state: ServiceStateSchema,
    detail: z.string().optional(),
    checkedAt: z.iso.datetime({ offset: true }),
  })
  .strict();

export type ComponentStatus = z.infer<typeof ComponentStatusSchema>;

export const SystemStatusSchema = z
  .object({
    service: z.string(),
    version: z.string(),
    state: ServiceStateSchema,
    checkedAt: z.iso.datetime({ offset: true }),
    components: z.array(ComponentStatusSchema),
  })
  .strict();

export type SystemStatus = z.infer<typeof SystemStatusSchema>;

export const ProviderCapabilitySchema = z
  .object({
    provider: z.string(),
    models: z.array(z.string()),
    supportsTools: z.boolean(),
    supportsStructuredOutput: z.boolean(),
  })
  .strict();

export type ProviderCapability = z.infer<typeof ProviderCapabilitySchema>;
